app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/sms/DailyReceivedWorkerTask.java

```diff
@@ -42,25 +41,91 @@ protected void onPostExecute(TreeMap<Integer, Integer> ret) {
         ArrayList<String> receivedSms = new ArrayList<>();
 
         for (int i = 0; i < list.size(); i++) {
-            if (list.get(i).getTime().equals(Calendar.getInstance())) {
+            String p = list.get(i).getType();
+            if (p.equals("1")) {
                 receivedSms.add(list.get(i).getTime());
             }
         }
         dailyReceivedText = getDailyTexts(receivedSms);
         return dailyReceivedText;
     }
 
+    // 12am=0, 3am=1, 6am=2 ,9am=3,12pm=4, (15)3pm=5, (18)6pm=6, (21)9pm=7, (24)12am=8
+
     private TreeMap<Integer, Integer> getDailyTexts(ArrayList<String> list) {
         for (int i = 0; i < list.size(); i++) {
             long lg = Long.parseLong(list.get(i));
             DateTime mDateTime = new DateTime(lg);
             int hourOfDay = mDateTime.getHourOfDay();
 
-            if (dailyReceivedText.containsKey(hourOfDay)) {
-                dailyReceivedText.put(hourOfDay, dailyReceivedText.get(hourOfDay) + 1);
-                dailyReceivedText.entrySet();
+
+            if (hourOfDay == 0) {
+                if (dailyReceivedText.containsKey(hourOfDay)) {
```

Comment:
combine {find,into}GetDailyTexts() into one method and check the map in that method